packages/libs/error-stack/src/fmt/mod.rs

```diff
@@ -99,29 +105,29 @@
 //! #     ansi_to_html::convert_escaped(value.as_ref()).unwrap()
 //! # }
 //! #
-//! # #[cfg(nightly)]
+//! # #[cfg(rust_1_65)]
 //! # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__doc.snap")].assert_eq(&render(format!("{report:?}")));
 //! #
 //! println!("{report:?}");
 //!
-//! # #[cfg(nightly)]
+//! # #[cfg(rust_1_65)]
 //! # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt_doc_alt.snap")].assert_eq(&render(format!("{report:#?}")));
 //! #
 //! println!("{report:#?}");
 //! ```
-//! ### `println!("{report:?}")`
+//!
+//! The output of `println!("{report:?}")`:
 //!
 //! <pre>
 #![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__doc.snap"))]
 //! </pre>
 //!
-//! ### `println!("{report:#?}")`
+//! The output of `println!("{report:#?}")`:
 //!
 //! <pre>
 #![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt_doc_alt.snap"))]
```

Comment:
Not requesting changes here, but the stack output is
```
- Warning...
- Info...
- Backtrace...
```
however, the snippets lists backtrace first, then the warning. Do you think it makes sense to reverse the order for snippets as well?

Reply:
Nice catch. I overlooked that one. I think it makes sense to reverse the snippets for each debug as well.

Reply:
Wait no, bs, it's already reversed. I see what going on.

Reply:
These are our processing steps:

Frame Stack:

```
Warning
Backtrace
ContextA
```

Reordered to:

```
ContextA
	Warning
	Backtrace
```

Reverse all attachments:

```
ContextA
	Backtrace
	Warning
```

Hooks on everything

```
ContextA (emit = vec![], snippets = vec![])
	Backtrace (emit = vec![Emit::defer(bt)], snippets = vec![btSnippet])
	Warning (emit = vec![Emit::next(warn)], snippets = vec![warnSnippet])
```

Render

```
ContextA Display
	Emit::next(warn)
	Emit::defer(bt)
	
btSnippet
warnSnippet
```

Reply:
Basically: We have the concept of defer for emits, but not for snippets. In theory, we could pair emit and snippets so that you could emit and then attach snippets to that emit instead, which would make them also deferred, but that doesn't sound like a nice API and complicated things.

e.g. `vec![Emit::defer(...).attach_snippet(...)]`

Reply:
I was just wondering if we should order the snippets the exact same way as we do with the trace. For `Backtrace` it's easy to identify it, but if you would attach multiple warnings, would this result in this?

```
Context
  - Warning A
  - Warning B
  
--
Warning B Snippet
Warning A Snippet
```

Reply:
No, the result would still be:

```
Context
  - Warning A
  - Warning B
  
--
Warning A Snippet
Warning B Snippet
```

This only happens on "defer" of emits.

Reply:
Assuming we `defer` the warning, the order in the stack would be reversed, but not in the snippets?
How often do we expect to use `defer`? Maybe we shouldn't reorder deferring at all. We made the assumption, that `backtrace` and `spantrace` should be the last entry, but what if users don't want this to happen? I think it would make the library more predictable without reordering the deferred

Reply:
moving conversation off GH

Reply:
deferred to upcoming PR

Reply:
To summarize: We will change the `Context` method slightly. Also, the `Emit` enumeration will not capture the emits. A hook will then look similar to this:
```rust
install_hook(|var, context| {
    context.emit(Suggestion(var));
    context.emit("do better next time :upside_down_face: ")
    context.snippet("My snippet goes here")
    Emit::Next
})
```

This has the advantage, that emits and snippets are both either `Emit::Next` or `Emit::Deferred`, so it can be ensured, that snippets and emits are in the same order.
Additionally, we probably will rename `Emit::Next` (maybe `Emit::Immediate`) and remove the reversal of `Emit::Deferred`